grpc/execution/test_utils.go

```diff
@@ -0,0 +1,169 @@
+package execution
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus"
+	beaconConsensus "github.com/ethereum/go-ethereum/consensus/beacon"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/params"
+	"math/big"
+	"testing"
+	"time"
+)
+
+var (
+	// testKey is a private key to use for funding a tester account.
+	testKey, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+
+	// testAddr is the Ethereum address of the tester account.
+	testAddr = crypto.PubkeyToAddress(testKey.PublicKey)
+
+	testBalance = big.NewInt(2e18)
+)
+
+func generateMergeChain(n int, merged bool) (*core.Genesis, []*types.Block, *ecdsa.PrivateKey, *ecdsa.PrivateKey) {
+	config := *params.AllEthashProtocolChanges
+	engine := consensus.Engine(beaconConsensus.New(ethash.NewFaker()))
+	if merged {
+		config.TerminalTotalDifficulty = common.Big0
+		config.TerminalTotalDifficultyPassed = true
+		engine = beaconConsensus.NewFaker()
+	}
+
+	bridgeAddressKey, err := crypto.GenerateKey()
+	if err != nil {
+		panic(err)
+	}
+	bridgeAddress := crypto.PubkeyToAddress(bridgeAddressKey.PublicKey)
+
+	config.AstriaRollupName = "astria"
+	config.AstriaSequencerInitialHeight = 10
+	config.AstriaCelestiaInitialHeight = 10
+	config.AstriaCelestiaHeightVariance = 10
+	config.AstriaBridgeAddressConfigs = []params.AstriaBridgeAddressConfig{
+		{
+			BridgeAddress:  bridgeAddress.Bytes(),
+			StartHeight:    2,
+			AssetDenom:     "nria",
+			AssetPrecision: 18,
+			Erc20Asset:     nil,
+		},
+	}
+
+	feeCollectorKey, err := crypto.GenerateKey()
+	if err != nil {
+		panic(err)
+	}
+	feeCollector := crypto.PubkeyToAddress(feeCollectorKey.PublicKey)
+
+	astriaFeeCollectors := make(map[uint32]common.Address)
+	astriaFeeCollectors[1] = feeCollector
+	config.AstriaFeeCollectors = astriaFeeCollectors
+
+	genesis := &core.Genesis{
+		Config: &config,
+		Alloc: core.GenesisAlloc{
+			testAddr:                         {Balance: testBalance},
+			params.BeaconRootsStorageAddress: {Balance: common.Big0, Code: common.Hex2Bytes("3373fffffffffffffffffffffffffffffffffffffffe14604457602036146024575f5ffd5b620180005f350680545f35146037575f5ffd5b6201800001545f5260205ff35b6201800042064281555f359062018000015500")},
```

Comment:
just curious, what is this code?

Reply:
eip-4788 stuff, address where historical beacon roots are stored. but we can remove it. don't need it. i copied this over from another test.